### PR TITLE
add nav guard to protect /Entrepreneur from not Auth users

### LINKED ISSUE #150: add nav guard to prevent a user from accessing /entrepreneur route when not authenticated

### PR DESCRIPTION
**Pull request by cezaryskura**
[#close #150 ]

### CODE REVIEW

**Comment by petarblazevski** [comment]
I thought we are getting rid of `aws-amplify`?

**Comment by cezaryskura** [comment]
> I thought we are getting rid of `aws-amplify`?

oops.. did i miss something ?

**Comment by petarblazevski** [comment]
https://github.com/I-OWE-YOU/web-frontend/issues/112

But maybe just for the API calls

File: src/router/index.js [hunk 1]
```diff
@@ -34,6 +45,12 @@ const vueRoutes = [
     name: 'Entrepreneur',
     component: () =>
       import(/* webpackChunkName: "Entrepreneur" */ '@views/Entrepreneur.vue'),
+    beforeEnter: async (to, from, next) => {
+      const isAuthenicated = await checkIfUserLoggedIn()
+      if (!isAuthenicated) {
+        next({ path: routes.signup })
+      } else next()
```
Comment (IAfanasovMob) [inline src/router/index.js:52]: it would be great to cover else branch with curly brackets

File: src/views/SignUp.vue [hunk 2]
```diff
@@ -130,19 +131,13 @@ export default {
         })
 
         await Auth.signIn(this.email, this.password)
-        this.navigateToEntrepreneur()
+        AmplifyEventBus.$emit('authState', AuthStateValue.SIGNED_IN)
```
Comment (IAfanasovMob) [inline src/views/SignUp.vue:134]: I didn't get how the navigation to the company address page would be triggered. in other world, what is responsible for navigating the user from singup page to "enter your address page"

Comment (cezaryskura) [reply src/views/SignUp.vue:134]: `AmplifyEventBus.$on('authState', async (state) => {
  switch (state) {
    case AuthStateValue.SIGNED_OUT:
      router.push({ path: routes.home })
      break
    case AuthStateValue.SIGNED_IN:
      router.push({ path: routes.entrepreneur })
      break

    default:
      break
  }
})`

Comment (cezaryskura) [reply src/views/SignUp.vue:134]: main.js

File: src/router/index.js (hunk 1 above)
Comment (cezaryskura) [reply src/router/index.js:52]: sorry, missed it 
done

**Pull request merged**